clicache/src/CacheFactory.cpp

```diff
@@ -60,12 +60,10 @@ namespace Apache
 				bool pdxIgnoreUnreadFields = false;
         bool pdxReadSerialized = false;
 				bool appDomainEnable = false; 
-        std::shared_ptr<native::Cache> nativeCache = nullptr;
         _GF_MG_EXCEPTION_TRY2
           //msclr::lock lockInstance(m_singletonSync);
           DistributedSystem::acquireDisconnectLock();
-    
-          nativeCache = m_nativeptr->get()->create( );
+          auto nativeCache = std::make_shared<native::Cache>(m_nativeptr->get()->create( ));
```

Comment:
I gave this a quick try, but I might call it out of scope.  Problem is that `Cache^` passes it's `shared_ptr` around by `const &` in several places to both managed/native sides.  It feels like it might explode out this pull request.  If it's alright with you I'd like to address it outside this pull request.